### ISSUE TITLE
Pointer being freed was not allocated [Git.Branch.name(ref)]

### ISSUE THREAD

**Issue by devrelm** [nodegit/nodegit#1012]
I'm getting an issue when calling `git.Branch.name(ref)`.

Running OS X 10.11.4, git version 2.8.1 (not sure if necessary.)
EDIT: Also, running node v0.12.13

gist: https://gist.github.com/devrelm/873f2109d8db6ecc39cd8b9a41645bce

```
node(99001,0x7fff757a3000) malloc: *** error for object 0x101f06817: pointer being freed was not allocated
*** set a breakpoint in malloc_error_break to debug
[1]    99001 abort      node test.js
```

## To Reproduce:

1. `git clone git@gist.github.com:873f2109d8db6ecc39cd8b9a41645bce.git`
2. `cd 873f2109d8db6ecc39cd8b9a41645bce`
3. `npm install nodegit`
4. `node test.js`

**Comment by johnhaley81** [comment]
Dat repro case. <3 I'll check this out soon.

**Comment by johnhaley81** [comment]
This seems to be fixed in 0.13.1. Thanks @devrelm.
